### DISCUSSION TITLE
tr/integrations/tensorrt/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#12075] [Docs]
# tr/integrations/tensorrt/

NVIDIA GPU'larda gelişmiş performans ve verimlilik için Ultralytics YOLOv8  modellerini TensorRT formatına aktarmanın gücünü ve esnekliğini keşfedin.

https://docs.ultralytics.com/tr/integrations/tensorrt/

**Comment by nyperu** [comment]
Please provide me with a text outlining compatibility of TensorRT with YOLOv8, CUDA, and necessary libraries versions for image processing.

**Reply by pderrenger** [reply]
Here's a brief outline on the compatibility of YOLOv8 with TensorRT, CUDA, and related library versions essential for processing images:

1. **TensorRT Compatibility**:
   - YOLOv8 models can be exported to the TensorRT format to leverage optimized inference on NVIDIA GPUs. This conversion boost efficiency and speed, particularly for real-time applications like object detection.
   
2. **CUDA and Library Dependence**:
   - For optimal TensorRT operations, you are generally required to use the latest or compatible CUDA version supported by your NVIDIA drivers and the TensorRT version.

3. **Necessary Libraries**:
   - Besides CUDA, TensorRT utilization with YOLOv8 will require the following libraries: `cuDNN` (a GPU-accelerated library for deep neural networks), `nvcc` (CUDA Compiler) for compiling model layers, along with standard image processing libraries like `OpenCV` for handling image inputs and manipulations.

Here's an example of how to export a YOLOv8 model to TensorRT format using the Ultralytics framework:

```python
from ultralytics import YOLO

model = YOLO("yolov8n.pt")

# Export to TensorRT format
model.export(format="engine", imgsz=640, dynamic=True)
```

This brief overview should help you integrate YOLOv8 models with TensorRT and CUDA effectively. If you need further details, please refer to the specific sections of the Ultralytics documentation or consider consulting the official NVIDIA documentation relevant to the TensorRT and CUDA versions you're employing.
